contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -1487,7 +1487,9 @@ is_tsql_char_type_with_len(Oid type)
 	return utilptr->is_tsql_bpchar_datatype(type) ||
 			utilptr->is_tsql_nchar_datatype(type) ||
 			utilptr->is_tsql_varchar_datatype(type) ||
-			utilptr->is_tsql_nvarchar_datatype(type);
+			utilptr->is_tsql_nvarchar_datatype(type) ||
+			utilptr->is_tsql_text_datatype(type) ||
+			utilptr->is_tsql_ntext_datatype(type);
```

Comment:
Why are we adding [N]TEXT here and not letting PG to decide the common type for these types? I assume that, PG will have different precedence and we don't want PG to decide that. Lets add proper comment in code to explain that.

Were we observing the similar issues for UNION as well? Lets add few testcases for UNION to test that.

Reply:
Added these conditions for CASE only (separate function), So no need to add union tests.